Mlem/App/Views/Root/Onboarding/SignUpView+Logic.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SignUpView+Logic.swift
+//  Mlem
+//
+//  Created by Sjmarf on 06/09/2024.
+//
+
+import MlemMiddleware
+import SwiftUI
+
+extension SignUpView {
+    var canSubmit: Bool {
+        guard let instance = instance as? any Instance2Providing else { return false }
+        return (!instance.captchaEnabled || !captchaAnswer.isEmpty)
```

Comment:
*micro-nit*

This might just be personal preference/my love of DeMorgan's laws, but I think the logic is a little clearer structured this way? It's pretty clear what the check is doing either way.

```suggestion
        return !(instance.captchaEnabled && captchaAnswer.isEmpty)
```

Reply:
Sure 👍